src/test/java/org/motechproject/prevac/service/CapacityInfoServiceTest.java

```diff
@@ -0,0 +1,158 @@
+        when(unscheduledVisitDataService.countFindByClinicIdAndDateRange(secondClinic.getId(), dateRange)).thenReturn(1L);
+
+        when(clinicDataService.count()).thenReturn(2L);
+
+        List<CapacityInfoDto> capacityInfoDtos = new ArrayList<>(Arrays.asList(
+                new CapacityInfoDto("first", 10, -5, 4, -1),
+                new CapacityInfoDto("second", 20, 9, -2, 16)));
+
+        Records<CapacityInfoDto> result = capacityInfoService.getCapacityInfoRecords(gridSettings);
+        for (int i = 0; i < capacityInfoDtos.size(); i++) {
+            checkIfCapacityDtoAreSame(capacityInfoDtos.get(i), result.getRows().get(i));
+        }
+    }
+
+    @Test
+    public void shouldReturnZerosForEmptyDateRange() {
+        GridSettings gridSettings = createGridSettings(1, 10, null, "", "");
+        List<Clinic> clinics = new ArrayList<>(Arrays.asList(
+                createClinic(1L, "siteId1", "first", 20, 5, 4, 3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0),
+                createClinic(2L, "siteId2", "second", 20, 10, 2, 9, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0)));
+
+        when(clinicDataService.retrieveAll(Mockito.any(QueryParams.class))).thenReturn(clinics);
+        when(clinicDataService.count()).thenReturn(2L);
+
+        List<CapacityInfoDto> capacityInfoDtos = new ArrayList<>(Arrays.asList(
+                new CapacityInfoDto("first", 0, 0, 0, 0),
+                new CapacityInfoDto("second", 0, 0, 0, 0)));
+
+        Records<CapacityInfoDto> result = capacityInfoService.getCapacityInfoRecords(gridSettings);
+        for (int i = 0; i < capacityInfoDtos.size(); i++) {
+            checkIfCapacityDtoAreSame(capacityInfoDtos.get(i), result.getRows().get(i));
+        }
+    }
+
+    private GridSettings createGridSettings(int page, int rows, DateFilter dateFilter, String startDate, String endDate) {
+        GridSettings gridSettings = new GridSettings();
+        gridSettings.setPage(page);
+        gridSettings.setRows(rows);
+        gridSettings.setDateFilter(dateFilter);
+        gridSettings.setStartDate(startDate);
+        gridSettings.setEndDate(endDate);
+        return gridSettings;
+    }
+
+    private void checkIfCapacityDtoAreSame(CapacityInfoDto expected, CapacityInfoDto result) {
+        assertEquals(expected.getClinic(), result.getClinic());
+        assertEquals(expected.getMaxCapacity(), result.getMaxCapacity());
+        assertEquals(expected.getAvailableCapacity(), result.getAvailableCapacity());
+        assertEquals(expected.getScreeningSlotRemaining(), result.getScreeningSlotRemaining());
+        assertEquals(expected.getVaccineSlotRemaining(), result.getVaccineSlotRemaining());
+    }
+
+    private Clinic createClinic(Long id, String siteId, String location, Integer numberOfRooms, Integer maxCapacityByDay, Integer maxScreeningVisits, //NO CHECKSTYLE ParameterNumber
```

Comment:
do you really need to set all of the fields here, most of them are set to 0, so just remove those params